src/test/java/jenkins/plugins/openstack/JcascTest.java

```diff
@@ -105,6 +105,7 @@ public void configure() {
             assertEquals("/tmp/foo", co.getFsRoot());
             assertEquals(((Integer) 42), co.getRetentionTime());
             LauncherFactory.SSH lf = (LauncherFactory.SSH) co.getLauncherFactory();
+            assertThat(co.getNodeProperties(), empty());
```

Comment:
This will be `null` after the suggested change.

Reply:
This definitely needs test coverage for declaring realistic node properties at JCasC level

Reply:
OK, understood ... but I'll hold off making any additions here until we've sorted out the null/empty node properties issue on SlaveOptions.

Reply:
Initial tests added as part of https://github.com/jenkinsci/openstack-cloud-plugin/pull/270/commits/f69471d6380ee1ddea2cefcc61b000c95f03484e
Note: Those changes don't test our ability to specify an empty list of node properties to override the defaults, and I'm not 100% sure of what the jcasc syntax would be for that...
My guess is that it'd be
```
          - name: "noNodeProperties"
            labels: "templateWithNoNodePropertiesInsteadOfDefaults"
            slaveOptions:
              nodeProperties: []
```
Would you agree?

Reply:
Yes, that is correct. The think is that everyone's favorite markup format, YAML, has no means of representing empty lists. So the fact YAML is a superset of JSON (yes, you can switch to JSON notation at will), comes to rescue here as JSON do have the empty list literal.

Reply:
Tests included in https://github.com/jenkinsci/openstack-cloud-plugin/pull/270/commits/5456d1d835b8da5588b9f7e3ce96730cc0b1a071